### PR TITLE
fix: avoid panic when ctx is canceled

### PR DESCRIPTION
**Pull request by yhl25**

### CODE REVIEW

**Review by vigith — APPROVED** [review]
how was it working so far?

File: pkg/udf/rpc/errors.go [hunk 1]
```diff
@@ -44,3 +44,8 @@ func (e ApplyUDFErr) IsInternalErr() bool {
 func (e ApplyUDFErr) Error() string {
 	return fmt.Sprint(e.Message)
 }
+
+// Is checks if the error is of the same type
+func (e ApplyUDFErr) Is(target error) bool {
```
Comment (whynowy) [inline pkg/udf/rpc/errors.go:49]: Where is the function used?

Comment (vigith) [reply pkg/udf/rpc/errors.go:49]: i think it is coming from "session" code.

Comment (vigith) [reply pkg/udf/rpc/errors.go:49]: https://github.com/numaproj/numaflow/blob/main/pkg/reduce/pnf/processandforward.go#L136

Comment (vigith) [reply pkg/udf/rpc/errors.go:49]: looks like it hit the never in `session` code :)

**Review by whynowy — APPROVED** [review]

**Pull request merged**